client/asset/btc/btc.go

```diff
@@ -2304,11 +2151,41 @@ func (btc *ExchangeWallet) run(ctx context.Context) {
 	}
 }
 
+// prepareRedemptionRequestsForBlockCheck prepares a copy of the
+// findRedemptionQueue, checking for missing block data along the way.
+func (btc *ExchangeWallet) prepareRedemptionRequestsForBlockCheck() []*findRedemptionReq {
+	// Search for contract redemption in new blocks if there
+	// are contracts pending redemption.
+	btc.findRedemptionMtx.RLock()
```

Comment:
It looks like `btc.trySetRedemptionRequestBlock(req)` can insert values and so the lock should be held for writes.